### PR TITLE
Remove vendor from the repo

### PR DESCRIPTION
**Pull request by earlgreyz**
Speeding up a build by ~1min is not worth making a mess.

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
+
+# Generate groups doesn't work with absolute paths so we need to hack it
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null \
```
[the remark this answers is not in the document]
Comment (earlgreyz) [reply hack/update-codegen.sh]: The script may change based on kubernetes version, I've added code-generator installation instructions to the developers guide.

File: docs/developer-guide.md [hunk 2]
```diff
@@ -3,51 +3,108 @@
...
+[ -z "$unvet" ] || vet_error
+
+echo >&2 " - Checking generated code"
+./hack/verify-codegen.sh > /dev/null || codegen_error
+```
+
 # Regenerating _pkg/client_ code
+First install kubernetes code-generator script, by running:
+```bash
+if [[ "$(go get k8s.io/code-generator 2>&1)" =~ "package k8s.io/code-generator: no Go files in .*" ]]; then
```
Comment (m-chrzan) [inline docs/developer-guide.md]: Running this as a script does download the required scripts but also outputs
```
script.sh: line 4: Error installing k8s.io/code-generator: No such file or directory
```
to `stderr`, exits with error code 127, and creates a file called `2` containing a single newline.

File: hack/update-codegen.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-chrzan) [reply hack/update-codegen.sh]: Note that the [OpenShift code generation example](https://github.com/openshift-evangelists/crd-code-generation) we are basing this on keeps vendor in the repo.

I'm starting to second-guess the decision to remove `vendor/`.

Another idea, though I'm not sure how I feel about it either, would be to keep non-golang dependencies (specifically, this bash generation script) in the in-repo `vendor/` directory, while using the standard `go dep` mechanism for pure golang libraries we depend on. Manually traversing the `GOPATH` seems *really* hacky to me.

**Comment by earlgreyz** [comment]
@m-chrzan as this is not a code dependency it would have to be manually added every time somebody uses `godep save ./...` as it's not a package used anywhere in the project it won't get saved (it probably got removed already without anyone noticing ;)). Secondly this is not even required you could write your code manually so it's rather a tool for developers and they can download it manually.

**Pull request merged**